enhancements/platforms/aws/customer-provisioned-vpc-and-subnets.md

```diff
@@ -0,0 +1,240 @@
+- Adding kubernetes.io/cluster/.*: shared to the user-provided subnets.  This will happen in Go immediately before calling out to Terraform to begin creating new resources.
+
+#### API
+The install-config YAML will grow one property on the AWS platform structure:
+- Subnets: a slice of subnet ID strings
+The install-config wizard will add two prompts:
+- A Select prompt:
+  - Message: VPC
+  - Help: The VPC to be used for installation.
+  - Default: installer-created
+  - Options: [“installer-created”] + results of a DescribeVpcs call.  Possibly checking tags and excluding VPCs which have kubernetes.io/cluster/.*: owned tags.
+- A MultiSelect prompt:
+  - Message: Subnets
+  - Help: The subnets to use for the cluster.
+  - Options: results of a DescribeSubnets call on the selected VPC.  Possibly checking tags and excluding subnets which have kubernetes.io/cluster/.\*: owned tags.
+
+#### Isolation between clusters
+
+Security group rules based on the VPC CIDR (e.g. machine-config server ingress) will be accessible to other clusters sharing the same VPC, and previous work considers the Ignition configs served by the MCS to be sensitive data.  Digging into the affected rules:
+- ICMP ingress.  We may be able to drop this completely.  I don’t see a need for it, and folks who want it can always go UPI ;).
+- TCP 22 ingress (SSH).  We can restrict this to control plane and compute security groups, and require folks setting up a SSH bastion to either add one of those security groups to their bastion instance or alter our rules to allow their bastion.
+- Control plane TCP 6443 ingress (Kubernetes API).  We can restrict this to the API load balancers.  Folks hitting localhost will not get out to the security group level, and everyone else should be coming in via the load balancers.
+- Control plane TCP 22623 ingress (MCS): this should be restricted to the control plane and compute security groups, instead of the current by-VPC-CIDR logic to avoid leaking sensitive Ignition configs to non-cluster entities sharing the VPC.
+
+
+### Risks and Mitigations
+
+#### Tagging subnets
+
+Currently Terraform does not support management of resources that already exist, so we cannot tag the subnets using Terraform’s aws_subnet resource or data_sources.  Instead, we will tag the subnets from Go immediately before applying the Terraform configuration.
+
+#### Destroy
+
+Destroy will remain mostly unchanged, hinging on the kubernetes.io/cluster/.\*: owned tag.  We will need to grow new code to store kubernetes.io/cluster/.\*: shared in metadata.json and remove it from resources on which it is found.
+
+We should probably revert:
+- #1268, which began removing instance profiles by name.  That was a workaround to recover from openshift-dev clusters which were partially-deleted by the DPP reaper.  Folks using the installer’s destroy code won’t need it, and while the risk of accidental name collision is low, I don’t think it’s worth taking that risk.
```

Comment:
based on the conversation in https://github.com/openshift/installer/pull/2461 seems like we are not going to revert 1268 for now.